lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -208,6 +208,8 @@ def create(self):
             else:
                 msg = "'%s' is currently deleting. Cannot create."
                 self.module.fail_json(msg=msg % self.name)
+        if self.zone is None:
```

Comment:
I prefer the treatment of self.cache_port on line 224

i.e move this to after the definition of `kwargs`, remove `PreferredAvailabilityZone` from `kwargs` and then do

```
if self.zone is not None:
   kwargs['PreferredAvailabilityZone'] = self.zone
```

Not really a blocker though